packages/runtime/runtime-definitions/src/id-compressor/idCompressor.ts

```diff
@@ -0,0 +1,135 @@
+	 * This includes finalized state as well as un-finalized state and is therefore suitable for use in offline scenarios.
+	 */
+	serialize(
+		withSession: boolean,
+	): SerializedIdCompressorWithOngoingSession | SerializedIdCompressorWithNoSession;
+
+	/**
+	 * Returns a persistable form of the current state of this `IdCompressor` which can be rehydrated via `IdCompressor.deserialize()`.
+	 * This includes finalized state as well as un-finalized state and is therefore suitable for use in offline scenarios.
+	 */
+	serialize(withSession: true): SerializedIdCompressorWithOngoingSession;
+
+	/**
+	 * Returns a persistable form of the current state of this `IdCompressor` which can be rehydrated via `IdCompressor.deserialize()`.
+	 * This only includes finalized state and is therefore suitable for use in summaries.
+	 */
+	serialize(withSession: false): SerializedIdCompressorWithNoSession;
+	serialize(withSession: boolean): SerializedIdCompressor;
+}
+
+/**
+ * This is the set of operations on an IdCompressor that DDS author
+ * would leverage. These functions allow for the generation of Ids and
+ * translation between different Id spaces.
+ */
+export interface IIdCompressor {
+	localSessionId: SessionId;
+
+	/**
+	 * Generates a new compressed ID or returns an existing one.
+	 * This should ONLY be called to generate IDs for local operations.
+	 * @param override - Specifies a specific string to be associated with the returned compressed ID.
+	 * Performance note: assigning override strings incurs a performance overhead.
+	 * @returns an existing ID if one already exists for `override`, and a new local ID otherwise. The returned ID is in session space.
+	 */
+	generateCompressedId(): SessionSpaceCompressedId;
+	/**
+	 * Normalizes a session space ID into op space.
+	 * @param id - the local ID to normalize.
+	 * @returns the ID in op space.
+	 */
+	normalizeToOpSpace(id: SessionSpaceCompressedId): OpSpaceCompressedId;
+	/**
+	 * Normalizes an ID into session space.
+	 * @param id - the ID to normalize. If it is a local ID, it is assumed to have been created by the session corresponding
+	 * to `sessionId`.
+	 * @param originSessionId - the session from which `id` originated
+	 * @returns the session-space ID corresponding to `id`, which might not have been a final ID if the client that created it had not yet
+	 * finalized it. This can occur when a client references an ID during the window of time in which it is waiting to receive the ordered
+	 * range that contained it from the server.
+	 */
+	normalizeToSessionSpace(
+		id: OpSpaceCompressedId,
+		originSessionId: SessionId,
+	): SessionSpaceCompressedId;
+
+	/**
+	 * Normalizes a final ID into session space.
+	 * @param id - the final ID to normalize.
+	 * @returns the session-space ID corresponding to `id`.
+	 */
+	normalizeToSessionSpace(id: FinalCompressedId): SessionSpaceCompressedId;
+	normalizeToSessionSpace(
```

Comment:
I don't think this signature should be on the interface, only the implementing class.

Reply:
Specifically, it should have one that takes a FinalCompressedId and one that takes an OpSpaceCompressedId and a non-optional SessionId. This last signature is the implementation signature.